doc/modules/linear_model.rst

```diff
@@ -61,6 +61,12 @@ example, when data are collected without an experimental design.
 
    * :ref:`sphx_glr_auto_examples_linear_model_plot_ols.py`
 
+Ordinary Least Squares uses a Singular Value Decomposition (SVD) based
```

Comment:
Would it be enough to explain the different solvers in the doc of the fit method?
Also, does the complexity section below need an update for cholesky (I think SVD and cholesky differ only by factors and lower order terms, so the big O notation still holds.)

Reply:
Are you suggesting I remove this from the `linear_model` documentation and only have it in the `fit` doc? And yes, I think the complexity analysis holds.

Reply:
I personally would move this solver section entirely to the `fit` documentation, though some core devs might have other opinions.

At least the complexity analysis below should be adjusted as the new solver options do not use SVD.

Reply:
> I personally would move this solver section entirely to the fit documentation, though some core devs might have other opinions.

We normally place this in the user guide and have the class docstring reference the user guide.  For example in [SVR](https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVR.html?highlight=svr#sklearn.svm.SVR.fit), we place the algorithm details in the user guide instead of `fit`.

My impression is that the `fit` docstring is not viewed frequently because of its position on the HTML rendered site.